FiniteCocompletions/examples/precompileLeftCartesianDistributivityFactoring_CompilerLogic.g

```diff
@@ -1,3 +1,19 @@
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "list", "length" ],
+        src_template := "List( [ 1 .. length ], i -> list[i] )",
+        dst_template := "list{[ 1 .. length ]}",
+    )
+);
+
+CapJitAddLogicTemplate(
+    rec(
+        variable_names := [ "entry1", "entry2", "entry3", "length" ],
+        src_template := "[ entry1, entry2, entry3 ]{[ 1 .. length ]}",
+        dst_template := "[ entry1, entry2, entry3 ]",
```

Comment:
This logic template looks really dangerous.

Reply:
Exactly: I do not understand why the compiler saved 3 in a global variable and was unable to match `[ entry1, entry2, entry3 ]{[ 1 .. 3 ]}`.

Reply:
Global variables usually come from attributes of categories or from context variables.

Reply:
But why can't the compiler match 

```gap
function ( T_1, source_1, a_1, b_1, c_1, target_1 )
    deduped_7_1 := [ a_1, b_1, c_1 ]{[ 1 .. CAP_JIT_INTERNAL_GLOBAL_VARIABLE_7 ]};
    ...
end
gap> CAP_JIT_INTERNAL_GLOBAL_VARIABLE_7;
3
```

Reply:
> But why can't the compiler match `CAP_JIT_INTERNAL_GLOBAL_VARIABLE_7 = 3`

Because the left hand side is an expression of type `EXPR_REF_GVAR` and the right hand is an expression of type `EXPR_INT`, and there is no rule matching expressions of different types. If you want to match `CAP_JIT_INTERNAL_GLOBAL_VARIABLE_7`, the following should work:
```
src_template := Concatenation( "[ entry1, entry2, entry3 ]{[ 1 .. ", CapJitGetOrCreateGlobalVariable( 3 )," ]}" ),
```
This creates a global variable which can be matched against `CAP_JIT_INTERNAL_GLOBAL_VARIABLE_7`.

Reply:
Wonderful, thank you very much. I adopted you suggestion :)